library/src/main/java/com/mapzen/android/OverlayManager.java

```diff
@@ -183,6 +207,146 @@ public MapData addMarker(Marker marker) {
         return addPointToMarkerMapData(marker);
     }
 
+    /**
+     * Draws two pins on the map. The start pin is active and the end pin is inactive.
+     * @param start
+     * @param end
+     */
+    public void drawRoutePins(LngLat start, LngLat end) {
+        if (startPinData == null) {
+            startPinData = mapController.addDataLayer(NAME_START_PIN);
+        }
+        if (endPinData == null) {
+            endPinData = mapController.addDataLayer(NAME_END_PIN);
+        }
+        startPinData.addPoint(start, null);
+        endPinData.addPoint(end, null);
+        mapController.requestRender();
```

Comment:
We can always merge as is and remove these later if it turns out they are no longer needed. Its good for the Tangram team to know the behavior is slightly different between devices too. I am testing on the Nextbit Robin (Android 6.0.1)